test_util/testharness.h

```diff
@@ -78,5 +78,40 @@ ::testing::AssertionResult AssertStatus(const char* s_expr, const Status& s);
 #define EXPECT_OK(s) \
   EXPECT_PRED_FORMAT1(ROCKSDB_NAMESPACE::test::AssertStatus, s)
 #define EXPECT_NOK(s) EXPECT_FALSE((s).ok())
+
+// std::regex has terrible performance in some cases, including possible stack
+// overflow, so should not be used in production. See
+// https://gcc.gnu.org/bugzilla/show_bug.cgi?id=61582 for example.
+// However, it is useful in testing. These wrappers facilitate use in testing
+// without triggering Facebook linters.
+class Regex {
+ public:
+  /*implicit*/ Regex(const std::string& pattern);
+  /*implicit*/ Regex(const char* pattern);
+  Regex(const Regex& other);
```

Comment:
That might encourage more production uses, which could be a bad thing to encourage. If we implement use of RE2 as an alternative to std::regex, it might be OK, but I'd prefer not to implement that now nor debate the issues here and now. This PR makes such a future endeavor easier.

Reply:
I thought the framework cleaned up these directories on shutdown but apparently that's just dbname_. I can fix.